### PR TITLE
:sparkles:Change where-resolver and placement-translator to use space framework

### PR DESCRIPTION
**Pull request by andreyod**
## Summary
Change where-resolver and placement-translator to use space framework and space-aware client instead of directly using workspaces.

## Related issue(s)
https://github.com/kubestellar/kubestellar/issues/853

Fixes #

### CODE REVIEW

**Comment by effi-ofer** [comment]
Dependent on issue #852

File: cmd/kubestellar-where-resolver/options/options.go [hunk 1]
```diff
@@ -38,17 +45,19 @@ func NewOptions() *Options {
 	logs.Config.Verbosity = config.VerbosityLevel(2)
 
 	return &Options{
+		SpaceMgtOpts:   *clientoptions.NewClientOpts("space-mgt", "access to space management workspaces"),
```
Comment (MikeSpreitzer) [inline cmd/kubestellar-where-resolver/options/options.go:48]: "workspaces" plural?

File: cmd/kubestellar-where-resolver/options/options.go [hunk 2]
```diff
@@ -38,17 +45,19 @@ func NewOptions() *Options {
...
 		BaseClientOpts: *clientoptions.NewClientOpts("base", "access to all logical clusters as kcp-admin"),
 		Logs:           logs,
+		Provider:       defaultProviderName,
+		KcsName:        defaultKcsSpaceName,
 	}
 }
 
 func (options *Options) AddFlags(fs *pflag.FlagSet) {
+	options.SpaceMgtOpts.SetDefaultUserAndCluster("kcp-admin", "root:space-mgt")
+	options.SpaceMgtOpts.AddFlags(fs)
```
Comment (MikeSpreitzer) [inline cmd/kubestellar-where-resolver/options/options.go:59]: What about the other fields that this PR adds to `Options`?

File: cmd/kubestellar-where-resolver/cmd/where-resolver-cmd.go [hunk 3]
```diff
@@ -70,6 +70,9 @@ func NewResolverCommand() *cobra.Command {
 		},
 	}
 
+	resolverCommand.Flags().StringVar(&options.Provider, "space-provider", options.Provider, "the name of the KubeStellar space provider")
+	resolverCommand.Flags().StringVar(&options.KcsName, "kcs-name", options.KcsName, "the name of the KubeStellar core space")
```
Comment (MikeSpreitzer) [inline cmd/kubestellar-where-resolver/cmd/where-resolver-cmd.go:74]: These should be part of `options.AddFlags`, right?

**Comment by MikeSpreitzer** [comment]
There's #929 too.

File: cmd/placement-translator/main.go [hunk 4]
```diff
@@ -131,10 +139,21 @@ func main() {
 		}
 	}()
 
-	espwRestConfig, err := espwClientOpts.ToRESTConfig()
```
Comment (MikeSpreitzer) [inline cmd/placement-translator/main.go:134]: Why is `espwClientOpts` retained in this PR?

File: cmd/placement-translator/main.go [hunk 5]
```diff
@@ -238,7 +256,7 @@ func main() {
 	// TODO: more
 	pt := placement.NewPlacementTranslator(concurrency, ctx, locationClusterPreInformer, epClusterPreInformer, spsClusterPreInformer, syncfgClusterPreInformer, customizerClusterPreInformer,
 		mbwsPreInformer, kcpClusterClientset, discoveryClusterClient, crdClusterPreInformer, bindingClusterPreInformer,
-		dynamicClusterClient, edgeClusterClientset, nsClusterPreInformer, nsClusterClient)
+		dynamicClusterClient, edgeClusterClientset, nsClusterPreInformer, msclient)
```
Comment (MikeSpreitzer) [inline cmd/placement-translator/main.go:259]: I still see some ...ClusterClients here.  These are signs of still using kcp workspaces.

**Pull request closed without merging**